lib/options.go

```diff
@@ -21,10 +21,97 @@
 package lib
 
 import (
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+
 	"github.com/loadimpact/k6/stats"
 	"gopkg.in/guregu/null.v3"
 )
 
+type TLSVersion struct {
+	Min int
+	Max int
+}
+
+func (v *TLSVersion) UnmarshalJSON(data []byte) error {
+	// From https://golang.org/pkg/crypto/tls/#pkg-constants
+	versionMap := map[string]int{
+		"ssl3.0": tls.VersionSSL30,
+		"tls1.0": tls.VersionTLS10,
+		"tls1.1": tls.VersionTLS11,
+		"tls1.2": tls.VersionTLS12,
+	}
+
+	// Version might be a string or an object with separate min & max fields
+	var fields struct {
+		Min string `json:"min"`
+		Max string `json:"max"`
+	}
+	if err := json.Unmarshal(data, &fields); err != nil {
+		switch err.(type) {
+		case *json.UnmarshalTypeError:
+			// Check if it's a type error and the user has passed a string
+			var version string
+			if otherErr := json.Unmarshal(data, &version); otherErr != nil {
+				switch otherErr.(type) {
+				case *json.UnmarshalTypeError:
+					return errors.New("Type error: the value of tlsVersion " +
+						"should be an object with min/max fields or a string")
+				}
+
+				// Some other error occurred
+				return otherErr
+			}
+			// It was a string, assign it to both min & max
+			fields.Min = version
+			fields.Max = version
+		default:
+			return err
+		}
+	}
+
+	var minVersion int
+	var maxVersion int
+	var ok bool
+	if minVersion, ok = versionMap[fields.Min]; !ok {
```

Comment:
Instead of all this, just do:

```go
if min, ok := versionMap[fields.Min]; ok {
    v.Min = min
}
```

Or if you want to avoid corrupting `v` in case of invalid data, construct a new TLSVersion at the top of the function, do that, and end it with a `*v = var`.